ui/lib/ldap/addon/components/page/role/create-and-edit.ts

```diff
@@ -0,0 +1,83 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+
+import type LdapRoleModel from 'vault/models/ldap/role';
+import { Breadcrumb, ValidationMap } from 'vault/vault/app-types';
+import type FlashMessageService from 'vault/services/flash-messages';
+import type RouterService from '@ember/routing/router-service';
+
+interface Args {
+  model: LdapRoleModel;
+  breadcrumbs: Array<Breadcrumb>;
+}
+interface RoleTypeOption {
+  title: string;
+  icon: string;
+  description: string;
+  value: string;
+}
+
+export default class LdapCreateAndEditRolePageComponent extends Component<Args> {
+  @service declare readonly flashMessages: FlashMessageService;
+  @service declare readonly router: RouterService;
+
+  @tracked modelValidations: ValidationMap | null = null;
+  @tracked invalidFormMessage = '';
+  @tracked error = '';
+
+  get roleTypeOptions(): Array<RoleTypeOption> {
+    return [
+      {
+        title: 'Static role',
+        icon: 'user',
+        description: 'Static roles map to existing users in an LDAP system.',
+        value: 'static',
+      },
+      {
+        title: 'Dynamic role',
+        icon: 'folder-users',
+        description: 'Dynamic roles allow Vault to create and delete a user in an LDAP system.',
+        value: 'dynamic',
+      },
+    ];
+  }
+
+  @task
+  @waitFor
+  *save(event: Event) {
+    event.preventDefault();
+
+    const { model } = this.args;
+    const { isValid, state, invalidFormMessage } = model.validate();
+
+    this.modelValidations = isValid ? null : state;
+    this.invalidFormMessage = isValid ? '' : invalidFormMessage;
+
+    if (isValid) {
+      try {
+        yield this.args.model.save();
+        this.flashMessages.success(`Successfully created the role ${model.name}`);
```

Comment:
I think this message should change to read `created/updated` depending on whether or not the model is new?

Reply:
💯 good catch!